### PR TITLE
feat(tokens): rename design tokens per naming audit

### PR DESCRIPTION
**Pull request by cixzhang**
## Token Naming Audit — Full Rename + Codemod

Implements the finalized token names from the [naming review spreadsheet](https://docs.google.com/spreadsheets/d/1MrjgNrs9E51RIl2o-SjVVgmb6Qu-vopFCcveTLR4KuY/edit), and adds a codemod so consumers can auto-migrate.

### Summary

| Type | Count |
|------|-------|
| Tokens renamed | 32 |
| Tokens removed | 3 |
| Files updated | 96 |
| Codemod tests | 22 |

### Color Renames

**Semantic clarity:**
- `--color-accent` → `--color-primary`
- `--color-positive` → `--color-success`
- `--color-negative` → `--color-error`
- `--color-educational` → `--color-info`
- `*-deemphasized` → `*-muted` (all status tokens)

**Noun-first grammar (thing → modifier):**
- `--color-hover-overlay` → `--color-overlay-hover`
- `--color-focus-outline-*` → `--color-ring-focus-*`

**Border/divider consolidation:**
- `--color-divider` → `--color-border`
- `--color-divider-high-contrast` → `--color-border-strong`

**Effects:**
- `--color-glimmer` → `--color-skeleton`
- `--color-shadow-elevation` → `--color-shadow`

### Token Removals

| Removed | Replacement |
|---------|-------------|
| `--color-accent-text` | `--color-text-link` |
| `--color-text-placeholder` | `--color-text-secondary` |
| `--color-icon-tertiary` | `--color-icon-secondary` |

### Codemod

Consumers can auto-migrate with:

```bash
npx @xds/cli upgrade --from 0.0.4 --to 0.0.5 --apply
# or run just this codemod:
npx @xds/cli upgrade --codemod migrate-token-names --apply
```

The `migrate-token-names` codemod handles:
- String literals (`colorVars["--color-accent"]`)
- `var()` references (`"var(--color-divider)"`)
- Template literals (`` `var(--color-accent)` ``)
- `defineTheme` token objects
- Longest-match-first to prevent partial replacements

22 test cases covering all renames, edge cases, and partial match safety.

---

### CODE REVIEW

File: internal/vibe-tests/prompts/evaluator.md [hunk 1]
```diff
@@ -74,24 +74,24 @@ XDS uses specific CSS variable naming. **Any variable not matching these pattern
 
 **Valid CSS variable prefixes:**
 
-| Category      | Prefix                                         | Examples                                                     |
-| ------------- | ---------------------------------------------- | ------------------------------------------------------------ |
-| Colors        | `--color-*`                                    | `--color-surface`, `--color-text-primary`, `--color-accent`  |
-| Spacing       | `--spacing-*`                                  | `--spacing-0` through `--spacing-7`                          |
+| Category      | Prefix                                         | Examples                                                                                 |
+| ------------- | ---------------------------------------------- | ---------------------------------------------------------------------------------------- |
+| Colors        | `--color-*`                                    | `--color-surface`, `--color-text-primary`, `--color-primary`                             |
```
Comment (cixzhang) [inline internal/vibe-tests/prompts/evaluator.md]: This row seems not quite right.

**Comment by rubyycheung** [comment]
Context on naming positive and negative https://github.com/facebookexperimental/xds/issues/683
